webapp/src/components/App/Singlestudy/explore/Modelization/Areas/Thermal/Matrix.tsx

```diff
@@ -19,91 +19,87 @@ interface Props {
-    "Min gen modulation",
-  ];
-
-  const tsGenNames = [
-    "FO Duration",
-    "PO Duration",
-    "FO Rate",
-    "PO Rate",
-    "NPO Min",
-    "NPO Max",
-  ];
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
 
   const handleChange = (event: React.SyntheticEvent, newValue: number) => {
     setValue(newValue);
   };
 
+  ////////////////////////////////////////////////////////////////
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
   return (
     <Box
-      width="100%"
-      height="100%"
-      display="flex"
-      flexDirection="column"
-      justifyContent="flex-start"
-      alignItems="center"
+      sx={{
+        display: "flex",
+        width: 1,
+        height: 1,
+        flexDirection: "column",
+        justifyContent: "flex-start",
+        alignItems: "center",
+      }}
     >
-      <Tabs
-        sx={{ width: 1 }}
-        value={value}
-        onChange={handleChange}
-        aria-label="basic tabs example"
-      >
+      <Tabs sx={{ width: 1 }} value={value} onChange={handleChange}>
         <Tab label={t("study.modelization.clusters.matrix.common")} />
         <Tab label={t("study.modelization.clusters.matrix.tsGen")} />
         <Tab label={t("study.modelization.clusters.matrix.timeSeries")} />
+        <Tab label={t("study.modelization.clusters.matrix.fuelCost")} />
+        <Tab label={t("study.modelization.clusters.matrix.co2Cost")} />
       </Tabs>
       <Box
         sx={{
           display: "flex",
-          width: "100%",
-          height: "100%",
+          width: 1,
+          height: 1,
         }}
       >
-        {R.cond([
-          [
-            () => value === 0,
-            () => (
-              <MatrixInput
-                study={study}
-                url={`input/thermal/prepro/${areaId}/${clusterId}/modulation`}
-                computStats={MatrixStats.NOCOL}
-                title={t("study.modelization.clusters.matrix.common")}
-                columnsNames={commonNames}
-              />
-            ),
-          ],
-          [
-            () => value === 1,
-            () => (
-              <MatrixInput
-                study={study}
-                url={`input/thermal/prepro/${areaId}/${clusterId}/data`}
-                computStats={MatrixStats.NOCOL}
-                title={t("study.modelization.clusters.matrix.tsGen")}
-                columnsNames={tsGenNames}
-              />
-            ),
-          ],
-          [
-            R.T,
-            () => (
-              <MatrixInput
-                study={study}
-                url={`input/thermal/series/${areaId}/${clusterId}/series`}
-                computStats={MatrixStats.NOCOL}
-                title={t("study.modelization.clusters.matrix.timeSeries")}
-              />
-            ),
-          ],
-        ])()}
+        {value === 0 && (
```

Comment:
```jsx
const matrixData = [
  { url: "modulation",  titleKey: "common", columns:  COMMON_MATRIX_COLS },
  ...
];

...

{matrixData.map(({ url, titleKey, columns }) => (
  <MatrixInput
    key={url}
    study={study}
    computStats={MatrixStats.NOCOL}
    url={`input/thermal/prepro/${areaId}/${clusterId}/${url}`}
    title={t(`study.modelization.clusters.matrix.${titleKey}`)}
    columnsNames={columns}
  />
))
}
```

Reply:
thanks I thought about it too, but I don't think it's necessary to optimize here

Reply:
it's not an optimization for the speed, but for the code, to remove duplicate JSX

Reply:
I know, I dont think it's necessary but your point is valid I will update this